src/fabric/interval.rs

```diff
@@ -133,4 +141,20 @@ impl Interval {
             panic!()
         }
     }
+
+    pub fn joint_with(&self, Interval { alpha_index, omega_index, .. }: &Interval) -> Option<usize> {
+        if self.alpha_index == *alpha_index || self.alpha_index == *omega_index {
+            Some(self.alpha_index)
+        } else if self.omega_index == *alpha_index || self.omega_index == *omega_index {
+            Some(self.omega_index)
+        } else {
+            None
+        }
+    }
+}
+
+impl PartialEq for Interval {
+    fn eq(&self, other: &Self) -> bool {
+        self.key() == other.key()
+    }
```

Comment:
Hmm, I wonder if you're gonna run into problems with this. You're leaving a lot of fields out of this equality check, and you might get tripped up later thinking two things shouldn't be equal when they are.

Reply:
Good point.